lib/libpcsc-cpp/include/pcsc-cpp/pcsc-cpp.hpp

```diff
@@ -254,39 +254,14 @@ class SmartCard
 };
 
 /** Reader provides card reader information, status and gives access to the smart card in it. */
-class Reader
+struct Reader
 {
-public:
-    enum class Status {
-        UNAWARE,
-        IGNORE,
-        CHANGED,
-        UNKNOWN,
-        UNAVAILABLE,
-        EMPTY,
-        PRESENT,
-        ATRMATCH,
-        EXCLUSIVE,
-        INUSE,
-        MUTE,
-        UNPOWERED,
-        _
-    };
-
-    Reader(ContextPtr context, string_t name, byte_vector cardAtr, flag_set<Status> status);
-
     SmartCard::ptr connectToCard() const { return std::make_unique<SmartCard>(ctx, name, cardAtr); }
 
-    bool isCardInserted() const { return status[Status::PRESENT]; }
-
-    std::string statusString() const;
-
+    ContextPtr ctx;
```

Comment:
Let's make `ctx` private for better encapsulation.

Reply:
I do not follow

Reply:
Making `ctx` private and restoring the constructor prevents external code from accidentally modifying `ctx` and lets us control how `Reader` objects are constructed. Hiding or controlling access to internal state is the core idea behind encapsulation. Like this:

```c++
struct Reader
 {
    Reader(string_t readerName, byte_vector atr, bool cardPresent, ContextPtr context) :
        name(readerName), cardAtr(atr), isCardPresent(cardPresent), ctx(context) {}

     SmartCard::ptr connectToCard() const { return std::make_unique<SmartCard>(ctx, name, cardAtr); }
 
     const string_t name;
     const byte_vector cardAtr;
     const bool isCardPresent;

private:
    ContextPtr ctx;
 };
```

The order of arguments needs to changed in `listReaders()` as well.